src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,50 +1,209 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (head == null) {
+            head = newNode;
+        } else {
+            tail.next = newNode;
+            tail = newNode;
+        }
+
+        tail = newNode;
+        size++;
+
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIsIndexExist(index, size, "Invalid index");
+
+        if (index == size) {
+            add(value);
+            return;
+        }
+
+        if (index == 0) {
+            addAtBeginning(value);
+        } else {
+            addByIndex(value, index);
+        }
+    }
+
+    private Node<T> findNodeByIndex(int index) {
```

Comment:
please move all private methods below public or at least below all usages of this private method